src/system/SystemMutex.cpp

```diff
@@ -82,7 +82,7 @@ DLL_EXPORT Error Mutex::Init(Mutex & aThis)
 #else
         aThis.mFreeRTOSSemaphore = xSemaphoreCreateMutex();
 #endif
-        if (aThis.mFreeRTOSSemaphore == NULL)
+        if (aThis.mFreeRTOSSemaphore == nullptr)
```

Comment:
All of this weird atomic code is only needed if mutex init is called from many threads at once, which would be an error...

Reply:
https://github.com/project-chip/connectedhomeip/issues/7415
It would be resolved in seperate PR, thanks